docs/developersguide/testing-practices.rst

```diff
@@ -0,0 +1,72 @@
+Testing practices
+=================
+
+It is a good practice to add test cases to each Invenio module. Tests are
+normally organized in:
+
+1. Unit functional testing: test cases run against the Flask app defined in the
+   module, ofter as example app. They can be found in ``tests`` or
+   ``tests/unit`` package.
+
+2. End-to-end functional testing: run by launching a browser and a live server
+   on the test Flask app and executing test cases through `Selenium`_.
+
+Invenio uses `pytest`_ as test framework.
+
+.. _Selenium: http://www.seleniumhq.org/
+.. _pytest: https://pypi.python.org/pypi/pytest
+
+Unit functional testing
+-----------------------
+
+To be completed
+
+End-to-end functional testing
+-----------------------------
+
+To run end-to-end tests, the requirements are:
+
+- Invenio installed, with modules that are covered by the tests
+- an Elastic Search running instance
+- a Redis running instance
+- a web browser driver
+
+The database is instead created (if not existing) as SQLite instance when tests
```

Comment:
How about rephrasing it as: `SQLite database will be created (if it doesn't exist) when tests are launched.` ?